### PR TITLE
Feature :  Added markdown-toc package to auto-generate toc with husky pre commit hook.

### LINKED ISSUE #1234: The `INSTALLATION.md` file needs a newly generated table of contents
**Is your feature request related to a problem? Please describe.**
1. There are many missing table of contents (TOC) entries in the INSTALLATION.md file.
1. Many of the existing TOC entries are under an inappropriate parent heading

This increases the risk of installation instructions being overlooked.

**Describe the solution you'd like**
1. Add the missing table of contents (TOC) entries in the INSTALLATION.md file.
1. Adjust the resulting TOC entries under the appropriate parent headings
1. Ideally, the auto-generation of the TOC should be automatic as part of the `git` process

**Describe alternatives you've considered**
- N/A

**Approach to be followed (optional)**
- Consider using a markdown TOC autogenerator

**Additional context**
- N/A

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by aialok**
**What kind of change does this PR introduce?**
-  Added update-toc script to pre-commit hook
- This commit adds the "update-toc" script to the pre-commit hook in the .husky/pre-commit file. This script updates the table of contents in the INSTALLATION.md file using the markdown-toc package.

**Issue Number:**

Fixes #1234 

**Did you add tests for your changes?**
- No

**Snapshots/Videos:**
- Here is clip of how table of content is getting generated on committing the changes

[cinnamon-2023-12-25T140007+0530.webm](https://github.com/PalisadoesFoundation/talawa-admin/assets/66772290/b9c08a28-59ff-4cb2-8b9b-ed896f4b3102)

**If relevant, did you update the documentation?**

- No 

**Summary**

Add markdown-toc package for generating table of contents
This commit adds the "markdown-toc" package to the project's dependencies in the package.json file. This package is used to generate a table of contents in the INSTALLATION.md file.

**Does this PR introduce a breaking change?**

- No

**Other information**

- 

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

- Yes

### CODE REVIEW

**Review by palisadoes — CHANGES_REQUESTED** [review]
This should be done for all the markdown files in the riot directory. That would solve a universal problem. Please update.

**Comment by palisadoes** [comment]
Please fix the failing typescript tests

**Comment by aialok** [comment]
Hey @palisadoes ! I have added a new update-toc.js file where it will execute `  markdown-toc -i "${file}" --bullets "-"` for each `.md` file present in the `talawa-admin`. 
Here is clip how it is working :

[cinnamon-2023-12-25T224523+0530.webm](https://github.com/PalisadoesFoundation/talawa-admin/assets/66772290/aa5dc36e-9f11-4dbc-ae84-0a3619fe5d56)

If it look good I will  push the changes.

**Comment by palisadoes** [comment]
1. Yes, go ahead. 
2. Please merge your code with the latest upstream

**Review by palisadoes — CHANGES_REQUESTED** [review]
1. Why is this PR updating typescript and language JSON files?
2. Please don't include them in the PR. Do not delete them from the repo.

**Comment by aialok** [comment]
> 1. Why is this PR updating typescript and language JSON files?
> 2. Please don't include them in the PR. Do not delete them from the repo.

Those changes are not mine. It there because I have fetched the latest upstream which is committed by other.

**Comment by palisadoes** [comment]
Though they may not be your edits, they must not be included in this PR. We only want to review what was required in the source issue

File: .husky/pre-commit [hunk 1]
```diff
@@ -4,5 +4,6 @@
 npm run format:fix
 npm run lint:fix
 npm run typecheck
+npm run update-toc
```
Comment (palisadoes) [inline .husky/pre-commit:7]: Will this only add the TOC to the root directory files? You have removed the .github files from the PR, but you still haven't answered my question.

**Review by palisadoes — CHANGES_REQUESTED** [review]
See comment

File: .husky/pre-commit (hunk 1 above)
Comment (aialok) [reply .husky/pre-commit:7]: Yes, I am only adding a Table of Contents to the root directory. I haven't made any changes to the .github files, so I'm unsure why the .github folder would be included in the pull request. I apologize if I didn't understand your question correctly.

toc

**Comment by aialok** [comment]
@palisadoes , please have a look.

File: package.json [hunk 2]
```diff
@@ -79,7 +80,8 @@
     "format:check": "prettier --check \"**/*.{ts,tsx,json,scss,css}\"",
     "typecheck": "tsc --project tsconfig.json --noEmit",
     "prepare": "husky install",
-    "jest-preview": "jest-preview"
+    "jest-preview": "jest-preview",
+    "update-toc": "node update-toc.js"
```
Comment (palisadoes) [inline package.json:84]: This script should not be in the root directory. We have a scripts directory referenced elsewhere in this file that should be used.

**Review by palisadoes — CHANGES_REQUESTED** [review]
See comment

File: package.json (hunk 2 above)
Comment (aialok) [reply package.json]: Fixes.

**Comment by palisadoes** [comment]
1. This solution doesn't use a script
1. It'll also be added to the Git hooks
- https://github.com/PalisadoesFoundation/talawa-api/pull/1587#discussion_r1437370185

PTAL

**Comment by palisadoes** [comment]
Is this integrated with the git hooks?

**Comment by aialok** [comment]
> Is this integrated with the git hooks?

Yes, I have added command in husky pre commit hook.

**Comment by palisadoes** [comment]
I thought about the universality of this solution. Will it work for developers running on the most OS  platforms?

1. Windows
1. Mac
1. Linux

For example, the find command is Unix specific

**Comment by aialok** [comment]
> I thought about the universality of this solution. Will it work for developers running on the most OS platforms?
> 
> 1. Windows
> 2. Mac
> 3. Linux
> 
> For example, the find command is Unix specific

- Hey, I try running in windows environment its not working as some of the `UNIX` command is not supported by `Windows PowerShell` and `command prompt` . Either we can add this to GitHub actions workflow and run this command. or we can roll back to node script which I wrote. I tried running node script in both windows and Linux and its working fine.

- One more things which we can do is to write script for each platform.

- Node Scripts

Screenshot 2023-12-28 133158

- Bash Command 

Screenshot 2023-12-28 134200

**Comment by palisadoes** [comment]
1. Create a typescript file and it to a `scripts/githooks` directory. It'll help with standardization
1. The `scripts` directory already exists.
1. Move any other git hooks scripts there too and update the hook for the new location

**Comment by aialok** [comment]
> 1. Create a typescript file and it to a `scripts/githooks` directory. It'll help with standardization
> 2. The `scripts` directory already exists.
> 3. Move any other git hooks scripts there too and update the hook for the new location

- Can I use javascript file as other script for `testing` in js and if we are using js then we dont need to compile ts to js and then run it with node.
- It will be easy for us to run the script.

**Comment by palisadoes** [comment]
> > 1. Create a typescript file and it to a `scripts/githooks` directory. It'll help with standardization
> > 2. The `scripts` directory already exists.
> > 3. Move any other git hooks scripts there too and update the hook for the new location
> 
>     * Can I use javascript file as other script for `testing` in js and if we are using js then we dont need to compile ts to js and then run it with node.
> 
>     * It will be easy for us to run the script.

OK

**Comment by aialok** [comment]
@palisadoes , Please Have a look : )

> > > 1. Create a typescript file and it to a `scripts/githooks` directory. It'll help with standardization
> > > 2. The `scripts` directory already exists.
> > > 3. Move any other git hooks scripts there too and update the hook for the new location
> > 
> > 
> > ```
> > * Can I use javascript file as other script for `testing` in js and if we are using js then we dont need to compile ts to js and then run it with node.
> > 
> > * It will be easy for us to run the script.
> > ```
> 
> OK

@palisadoes , Please have a look : )

**Comment by aialok** [comment]
Hi @palisadoes , could you please check this out? Thanks!

**Review by palisadoes — APPROVED** [review]

**Pull request merged**